pilot test fails with 'no configuration has been provided' if KUBECONFIG is missing

Unset the `$KUBECONFIG` then running `make test` fails with the following error:
```
--- FAIL: TestGetAPIServerExtensionCACert (0.00s)
        admit_test.go:453: invalid configuration: no configuration has been provided
FAIL
FAIL    istio.io/istio/pilot/pkg/kube/admit     0.415s
?       istio.io/istio/pilot/pkg/kube/admit/testcerts   [no test files]
--- FAIL: TestInitializerRun (0.00s)
        initializer_test.go:43: invalid configuration: no configuration has been provided
--- FAIL: TestInitialize (0.00s)
        initializer_test.go:43: invalid configuration: no configuration has been provided
--- FAIL: TestGetMeshConfig (0.00s)
        initializer_test.go:43: invalid configuration: no configuration has been provided
--- FAIL: TestGetInitializerConfig (0.00s)
        initializer_test.go:43: invalid configuration: no configuration has been provided
FAIL
FAIL    istio.io/istio/pilot/pkg/kube/inject    0.498s
```

This happens because:
- `make setup` target creates an empty file at `pilot/pkg/kube/config`
- `tests/k8s/helper.go` returns this empty file as the k8s config file if `$KUBECONFIG` is not set (https://github.com/istio/istio/blob/master/tests/k8s/helper.go), since `os.Stat` returns success

--- a/Makefile
+++ b/Makefile
@@ -93,7 +93,7 @@ checkvars:
 	@if test -z "$(TAG)"; then echo "TAG missing"; exit 1; fi
 	@if test -z "$(HUB)"; then echo "HUB missing"; exit 1; fi
 
-setup: pilot/pkg/kube/config
+setup:
 
 #-----------------------------------------------------------------------------
 # Target: depend
@@ -532,7 +532,7 @@ artifacts: docker
 	@echo 'To be added'
 
 pilot/pkg/kube/config:
-	touch $@
+	@echo 'nothing needs to be done for' $@
 
 kubelink:
 	ln -fs ~/.kube/config pilot/pkg/kube/

--- a/Makefile
+++ b/Makefile
@@ -531,9 +531,6 @@ push: checkvars clean.installgen installgen
 artifacts: docker
 	@echo 'To be added'
 
-pilot/pkg/kube/config:
-	@echo 'nothing needs to be done for' $@
-
 kubelink:
 	ln -fs ~/.kube/config pilot/pkg/kube/
 

